nutanix/resource_access_control_policy.go

```diff
@@ -0,0 +1,755 @@
+package nutanix
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/spf13/cast"
+	v3 "github.com/terraform-providers/terraform-provider-nutanix/client/v3"
+	"github.com/terraform-providers/terraform-provider-nutanix/utils"
+	"strings"
+)
+
+func resourceNutanixAccessControlPolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNutanixAccessControlPolicyCreate,
+		Read:   resourceNutanixAccessControlPolicyRead,
+		Update: resourceNutanixAccessControlPolicyUpdate,
+		Delete: resourceNutanixAccessControlPolicyDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"api_version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"metadata": {
+				Type:     schema.TypeMap,
+				Computed: true,
+				Elem: &schema.Resource{
```

Comment:
We can use it as TypeMap since it is only computed, but having Elem on TypeMap will raise an error, there is a two options:
- Leave is a Map and removes Elem.
- if we need to have all the attributes explicit we have to change to TypeList. with MaxValue=1,


 any thoughts? @yannickstruyf3